Python/Product/TestAdapter.Executor/UnitTest/UnitTestTestDiscoverer.cs

```diff
@@ -138,16 +138,22 @@ ITestCaseDiscoverySink discoverySink
 
         public string[] GetArguments(IEnumerable<string> sources, PythonProjectSettings settings, string outputfilename) {
             var arguments = new List<string>();
-            arguments.Add(DiscoveryAdapterPath);
-            arguments.Add("discover");
-            arguments.Add("unittest");
-            arguments.Add("--output-file");
-            arguments.Add(outputfilename);
-            //Note unittest specific options go after this separator
-            arguments.Add("--");
-            arguments.Add(settings.UnitTestRootDir);
-            arguments.Add(settings.UnitTestPattern);
-
+            foreach (var source in sources.ToArray()) {
```

Comment:
You don't need call `ToArray()` on an `IEnumerable<string>` to enumerate it. Consider removing this.

Reply:
And yes, if there's always guaranteed to be one element, you can get rid of the loop and only use the single element.